test/jdk/javax/swing/JMenuBar/TestMenuMnemonic.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8326458
+ * @key headful
+ * @requires (os.family == "windows")
+ * @modules java.desktop/com.sun.java.swing.plaf.windows
+ * @summary Verifies if menu mnemonics toggle on F10 press in Windows LAF
+ * @run main TestMenuMnemonic
+ */
+
+import java.awt.event.KeyEvent;
+import java.awt.Robot;
+import java.util.concurrent.atomic.AtomicInteger;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuItem;
+import javax.swing.JMenuBar;
+import javax.swing.MenuElement;
+import javax.swing.MenuSelectionManager;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+import com.sun.java.swing.plaf.windows.WindowsLookAndFeel;
+
+public class TestMenuMnemonic {
+
+    private static JFrame frame;
+    private static JMenuBar menuBar;
+    private static JMenu fileMenu;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("com.sun.java.swing.plaf.windows.WindowsLookAndFeel");
+        int expectedMnemonicShowHideCount = 5;
+        Robot robot = new Robot();
+        robot.setAutoDelay(100);
+        AtomicInteger mnemonicHideCount = new AtomicInteger(0);
+        AtomicInteger mnemonicShowCount = new AtomicInteger(0);
+
+        try {
+            SwingUtilities.invokeAndWait(TestMenuMnemonic::createAndShowUI);
+            robot.waitForIdle();
+            robot.delay(1000);
+
+            for (int i = 0; i < 10; i++) {
```

Comment:
```suggestion
            for (int i = 0; i < EXPECTED * 2; i++) {
```
No magic numbers, after all the number of `EXPECTED` show/hide depends on how many iterations the loop does.